packages/manager/modules/iam/src/components/fieldSpinner/fieldSpinner.directive.js

```diff
@@ -0,0 +1,54 @@
+import './fieldSpinner.styles.scss';
+
+/**
+ * @directive field-spinner
+ * @directive field-spinner-options
+ *
+ * @description
+ *
+ * Display a spinner inside the element on which the directive applies
+ * only if the passed expression to the field-spinner attribute is truthy
+ *
+ * You can specify options inside the field-spinner-options attribute
+ * Options are
+ * - top  : a number in rem that replaces the default top css property
+ * - left : a number in rem that replaces the default left css property
+ * - size : the size of the spinner. Default is "s"
+ */
+
+const defaultOptions = {
+  top: 0,
+  left: 0,
+  size: 's',
+};
+
+export default /* @ngInject */ function($compile) {
+  const link = (scope, element, attributes) => {
+    const { fieldSpinner, fieldSpinnerOptions } = attributes;
+    const options = { ...defaultOptions, ...scope.$eval(fieldSpinnerOptions) };
+    const style = `{
+      ${[options.left, options.top]
+        .filter(Boolean)
+        .map((x) => `${x}rem`)
+        .join(',')}
+    }`;
+
+    const ouiSpinner = angular.element(`
+      <oui-spinner
+        data-ng-if="${fieldSpinner}"
+        data-ng-style="${style}"
+        data-size="${options.size}"
+      ></oui-spinner>
+    `);
+
+    element.addClass('iam-field-spinner');
+    element.append(ouiSpinner);
+
+    $compile(ouiSpinner)(scope);
+  };
+
+  return {
+    restrict: 'A',
+    link,
+  };
+}
```

Comment:
**nitpick:** Since this new directive adds the capability to place the spinner at a particular place. This extra capability can be achieved using bootstrap utility classes. Creating an extra wrapper isn't worth it.

Also, creating wrappers in the specific module will make it difficult to understand as each module can create wrappers in multiple ways. So working across the modules will be difficult. Avoid creating wrappers.